app/controllers/pieces_controller.rb

```diff
@@ -1,2 +1,27 @@
 class PiecesController < ApplicationController
+  def show
+    @game =  Game.find(params[:id])
+    @piece = Piece.find(params[:id])
+#not functional code just thinking through the process
+      first_selected_piece = @piece
+      first_selected_piece.update_attributes(x, y)
+    end
+    @piece.update_attributes.update_attributes(x, y)
+  end
+
+  def update
+    @piece = Piece.find(params:id)
+    x = params[:x_position]
```

Comment:
Since you have `piece_params`. Should be able to do the `update` like this

```Ruby
@piece = Piece.find(params:id)
@piece.update_attributes(piece_params)
```